stage1/wasi.c

```diff
@@ -5,6 +5,18 @@
 #include <string.h>
 #include <time.h>
 
+#if defined(__linux__)
+#include <elf.h>
+#include <sys/auxv.h>
+#include <sys/resource.h>
+
+#if ULONG_MAX == 0xffffffff
+typedef Elf32_Phdr Elf_Phdr;
+#else
+typedef Elf64_Phdr Elf_Phdr;
+#endif
+#endif
+
```

Comment:
this code was intentionally written to not depend on posix or os-specific things. This patch breaks that. Please stick to only the C language and standard libc in wasm2c.

Reply:
As I noted on https://github.com/ziglang/zig/issues/22111#issuecomment-2543503244 and on Zulip, the kernel's indifference to `PT_GNU_STACK` leaves us with not a lot of options.

If you'd prefer, we could `setrlimit(RLIMIT_STACK, ...)` just before `fork()`ing in `bootstrap.c`, which already has -- and will grow more -- platform-specific code. We already hardcode the stack size values in that file anyway, so perhaps that's more fitting.

Reply:
I think that's the right way to go :+1:

Reply:
Although one downside of that approach is that it won't help if you're using CMake.

Do we care? Are we getting rid of CMake once we get rid of the LLVM dependency?

Reply:
yes and it will be a glorious day